### PR TITLE
feat: add badge component

### PR DESCRIPTION
**Pull request by wralith**
#24 

If it wasn't tailwind, it would look elegant. I tried few things but css child selector seems much proper. Maybe I can group similar utility classes in one string for readability, or custom classes?

Usage will be:
```jsx
<Badge>Label</Badge> // simple

 // with title, title position depends on order of `Title` and `Content`
<Badge>
  <Badge.Title>Title</Badge.Title> 
  <Badge.Content>Content</Badge.Content>
</Badge>
```

BTW there is no constraints to not use child e.g.  `<Badge.Title>` outside of  `<Badge>` scope , but i don't think it will be a problem for this project.

### CODE REVIEW

File: components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+type BadgeBaseProps = VariantProps<typeof badgeStyles>;
+interface Props extends BadgeBaseProps {
+  children: ReactNode;
+  className?: string;
+}
+
+const Badge = ({ children, className, size, variant }: Props) => {
+  return (
+    <div className={badgeStyles({ className, size, variant })}>
+      {typeof children == 'string' ? <span>{children}</span> : children}
```
Comment (oguzergul) [inline components/Badge/Badge.tsx:99]: react handles this case. we don't need to check for type-of badge content.

File: components/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    </div>
+  );
+};
+
+const BadgeTitle = ({ children }: { children: string }) => {
+  return <h4 className="shrink-0 rounded-2xl">{children}</h4>;
```
Comment (oguzergul) [inline components/Badge/Badge.tsx:105]: I am not sure if we should use the h4 element for this. can you check for the semantic order of the badge component?

**Review by oguzergul — CHANGES_REQUESTED** [review]

File: components/Badge/Badge.tsx (hunk 1 above)
Comment (wralith) [reply components/Badge/Badge.tsx:99]: Purpose here is creating a child element *-if not provided-* to apply css rules onto.

File: components/Badge/Badge.tsx (hunk 2 above)
Comment (wralith) [reply components/Badge/Badge.tsx:105]: You are right the header level depends on context, fixed `h4` is a mistake. 

I searched few UI libraries, they implemented it either as div or span, what do you think it should be?

Comment (oguzergul) [reply components/Badge/Badge.tsx:105]: span is okay for me.

File: components/Badge/Badge.tsx (hunk 1 above)
Comment (oguzergul) [reply components/Badge/Badge.tsx:99]: it is okay for me than.

**Review by oguzergul — APPROVED** [review]

**Pull request merged**